### PR TITLE
[exporter/elasticsearchexporter]: Rollback to client 7.17.7 due to incompatibility 

### PR DESCRIPTION
**Pull request by leewoobin789**
**Description:** <Describe what has changed.>
Elasticsearch version 7.10 is incompatibility due to newer version of elasticsearch client. 

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16310

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by leewoobin789** [comment]
@canhnt fyi

**Comment by canhnt** [comment]
Thank you @leewoobin789  for working on this. 

As go-elasticsearch client is only [forward compatible](https://github.com/elastic/go-elasticsearch/blob/7.17/.doc/installation.asciidoc#es-version-compatibility), i.e. clients version 7.x work with servers 7.x, clients 8.x work with server 8.x, is it possible if the elasticsearchexporter can support both most popular server versions (7.x & 8.x)?

**Comment by leewoobin789** [comment]
@canhnt good point. we can make it switchable by adding an optional configuration field with default value of "v8".

**Comment by JaredTan95** [comment]
> @canhnt good point. we can make it switchable by adding an optional configuration field with default value of "v8".

yes, if you have time to contribute this, you can finish this with another PR.

**Comment by JaredTan95** [comment]
can you fix `checks` ci https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3491419745/jobs/5844100875#step:11:468?

**Comment by leewoobin789** [comment]
@JaredTan95 done. I will directly start off adding optional field

**Comment by JaredTan95** [comment]
> @JaredTan95 done. I will directly start off adding optional field

okay, waiting for your next submission.

File: exporter/elasticsearchexporter/elasticsearch_bulk.go [hunk 1]
```diff
@@ -20,26 +20,25 @@ import (
...
-	"errors"
 	"fmt"
 	"io"
-	"net"
 	"net/http"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
-	elasticsearch "github.com/elastic/go-elasticsearch/v8"
-	esutil "github.com/elastic/go-elasticsearch/v8/esutil"
```
Comment (MovieStoreGuy) [inline exporter/elasticsearchexporter/elasticsearch_bulk.go:32]:
```suggestion
	elasticsearch "github.com/elastic/go-elasticsearch/v7"
	esutil "github.com/elastic/go-elasticsearch/v7/esutil"
```

Unless you're explicitly importing two seperate versions of the library, then the numbering isn't required.

File: exporter/elasticsearchexporter/elasticsearch_bulk.go [hunk 2]
```diff
@@ -132,27 +131,6 @@ func newElasticsearchClient(logger *zap.Logger, config *Config) (*esClientCurren
...
-		if isNetError := errors.As(err, &netError); isNetError && netError != nil {
-			// on Timeout (Proposal: predefined configuratble rules)
-			if !netError.Timeout() {
-				shouldRetry = true
-			}
-		}
-
-		return shouldRetry
-	}
-}
```
Comment (MovieStoreGuy) [inline exporter/elasticsearchexporter/elasticsearch_bulk.go:155]: Can I ask why this is being removed from this file?

File: .chloggen/elastic-exporter-version-rollback.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: elasticsearchexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: roll back elasticsearch client to v7.17.7 due to incompatibility with older versions
```
Comment (MovieStoreGuy) [inline .chloggen/elastic-exporter-version-rollback.yaml:8]: Do you mean `newer` version here?

And is there an upstream issue also tracking this?

**Review by MovieStoreGuy — COMMENTED** [review]
Just a few things I'd like clarified.

File: exporter/elasticsearchexporter/elasticsearch_bulk.go (hunk 2 above)
Comment (leewoobin789) [reply exporter/elasticsearchexporter/elasticsearch_bulk.go:155]: This function is solely used for the client option field "retryOnError" which exists in v8

File: .chloggen/elastic-exporter-version-rollback.yaml (hunk 3 above)
Comment (leewoobin789) [reply .chloggen/elastic-exporter-version-rollback.yaml:8]: Goo point wording sound confusing. I meant older version of elaticsrarch. I will change it

Comment (leewoobin789) [reply .chloggen/elastic-exporter-version-rollback.yaml:8]: @MovieStoreGuy https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16310

File: exporter/elasticsearchexporter/elasticsearch_bulk.go (hunk 1 above)
Comment (leewoobin789) [reply exporter/elasticsearchexporter/elasticsearch_bulk.go:32]: i am aware of that it needs to be version-specifically imported as the goal is also to have both versions depending on optional config for client version

**Comment by leewoobin789** [comment]
> > @JaredTan95 done. I will directly start off adding optional field
> 
> okay, waiting for your next submission.

I would add the proposed optional field on a separate PR, as it would bring change in code structure.
I think there needs to be a clear specification of from which version Otel will support, considering that there is already Elasticsearch version 8.5

**Comment by leewoobin789** [comment]
@JaredTan95 @canhnt this client version works even on 7.8

File: exporter/elasticsearchexporter/elasticsearch_bulk.go (hunk 1 above)
Comment (MovieStoreGuy) [reply exporter/elasticsearchexporter/elasticsearch_bulk.go:32]: Just to clarify, you would like both version 7 and 8 to be part of this file?

File: exporter/elasticsearchexporter/elasticsearch_bulk.go (hunk 2 above)
Comment (MovieStoreGuy) [reply exporter/elasticsearchexporter/elasticsearch_bulk.go:155]: Your change sets has the imports version 7, wouldn't that mean that this function is still needed?

Comment (leewoobin789) [reply exporter/elasticsearchexporter/elasticsearch_bulk.go:155]: No. as the purpose of this PR is to roll back to v7. this function cannot be used anywhere in the client, as it is dedicatedly written for "retryOnError" callback field of v8. so it is better to remove it.

**Comment by MovieStoreGuy** [comment]
I suspect this may have issues when it comes to dependabot raising changes, you may want to exclude this version within the `go.mod` file.

@codeboten, would that be the right way to go about this?

File: exporter/elasticsearchexporter/utils_test.go [hunk 4]
```diff
@@ -50,7 +50,7 @@ type httpTestError struct {
 	cause   error
 }
 
-const currentESVersion = "8.4.0"
+const currentESVersion = "7.17.7"
```
Comment (dmitryax) [inline exporter/elasticsearchexporter/utils_test.go:53]: Does this PR still support v8 server? If so can you please update the tests to test again both versions?

Comment (leewoobin789) [reply exporter/elasticsearchexporter/utils_test.go:53]: @dmitryax  sorry for late response. yes it does. I have tested agains both of each very first and the most current version. the result was fine as expected.

**Review by JaredTan95 — CHANGES_REQUESTED** [review]
LGTM

pls resolve conflicts.

**Comment by JaredTan95** [comment]
label -stale

**Comment by SundownSpecial** [comment]
@leewoobin789 sorry for bothering you, could you please resolve conflicts?

**Comment by leewoobin789** [comment]
@JaredTan95 @mx-psi @ali-essam sorry for being late. merge conflict should be resolved by now.

**Review by JaredTan95 — APPROVED** [review]
LGTM

**Review by dmitryax — APPROVED** [review]

**Pull request merged**